xarray/core/accessor_str.py

```diff
@@ -865,49 +1061,838 @@ def replace(self, pat, repl, n=-1, case=None, flags=0, regex=True):
+            output_sizes={group_dim: pat.groups, match_dim: maxcount},
+        ).astype(self._obj.dtype.kind)
+
+    def findall(
+        self,
+        pat: Any,
+        case: bool = None,
+        flags: int = 0,
+    ) -> Any:
+        """
+        Find all occurrences of pattern or regular expression in the DataArray.
+
+        Equivalent to applying re.findall() to all the elements in the DataArray.
+        Results in an object array of lists.
+        If there is only one capture group, the lists will be a sequence of matches.
+        If there are multiple capture groups, the lists will be a sequence of lists,
+        each of which contains a sequence of matches.
+
+        Parameters
+        ----------
+        pat : str or re.Pattern
+            A string contain a regular expression or
+            a compiled regular expression object.
+        case : bool, default: True
+            If True, case sensitive.
+            Cannoy be set if `pat` is a compiled regular expression.
+        flags : int, default: 0
+            - re module flags, e.g. re.IGNORECASE. Use ``0`` for no flags.
+            - Cannot be set if `pat` is a compiled regex
+
+        Returns
+        -------
+        extracted : same type as values or object array
+
+        Raises
+        ------
+        ValueError : `pat` has no capture groups.
+        ValueError : `case` is set when `pat` is a compiled regular expression.
+
+        Examples
+        --------
+        Create a string array
+
+        >>> value = xr.DataArray(
+        ...     [
+        ...         [
+        ...             "a_Xy_0",
+        ...             "ab_xY_10-bab_Xy_110-baab_Xy_1100",
+        ...             "abc_Xy_01-cbc_Xy_2210",
+        ...         ],
+        ...         [
+        ...             "abcd_Xy_-dcd_Xy_33210-dccd_Xy_332210",
+        ...             "",
+        ...             "abcdef_Xy_101-fef_Xy_5543210",
+        ...         ],
+        ...     ],
+        ...     dims=["X", "Y"],
+        ... )
+
+        Extract matches
+
+        >>> value.str.findall(r"(\\w+)_Xy_(\\d*)")
+        <xarray.DataArray (X: 2, Y: 3)>
+        array([[list([('a', '0')]), list([('bab', '110'), ('baab', '1100')]),
+                list([('abc', '01'), ('cbc', '2210')])],
+               [list([('abcd', ''), ('dcd', '33210'), ('dccd', '332210')]),
+                list([]), list([('abcdef', '101'), ('fef', '5543210')])]],
+              dtype=object)
+        Dimensions without coordinates: X, Y
+
+        See Also
+        --------
+        DataArray.str.extract
+        DataArray.str.extractall
+        re.compile
+        re.findall
+        pandas.Series.str.findall
+        """
+        pat = self._re_compile(pat, flags, case)
+
+        if pat.groups == 0:
+            raise ValueError("No capture groups found in pattern.")
+
+        return self._apply(pat.findall, dtype=np.object_)
+
+    def partition(
+        self,
+        dim: Optional[Hashable],
```

Comment:
```suggestion
        dim: Optional[Hashable] = "parts",
```
I would also give a reasonable default here. I feel a DataArray of lists is not very useful.